Nil pointer dereference while running apiserver.TestMaxInFlight

Seen on [travis](https://travis-ci.org/GoogleCloudPlatform/kubernetes/jobs/68890871) today:
```
E0629 23:23:31.027522   12633 errors.go:56] apiserver received an error that is not an api.Status: test Error
E0629 23:23:31.760228   12633 apiserver.go:360] Failed to parse "not a timeout": time: invalid duration not a timeout
E0629 23:23:31.930853   12633 apiserver.go:319] apiserver was unable to write a JSON response: type apiserver.UnregisteredAPIObject is not registered for "version" and it will be impossible to Decode it, therefore Encode will refuse to encode it.
E0629 23:23:31.931037   12633 errors.go:56] apiserver received an error that is not an api.Status: type apiserver.UnregisteredAPIObject is not registered for "version" and it will be impossible to Decode it, therefore Encode will refuse to encode it.
--- FAIL: TestMaxInFlight-2 (1.07s)
	handlers_test.go:48: unexpected response: 429
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xb code=0x1 addr=0x10 pc=0x4fa1cb]
goroutine 707 [running]:
github.com/GoogleCloudPlatform/kubernetes/pkg/apiserver.expectHTTP(0xc2083389c0, 0x24, 0xc8, 0xc208078e10)
	/home/travis/gopath/src/github.com/GoogleCloudPlatform/kubernetes/_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/apiserver/handlers_test.go:47 +0x16b
github.com/GoogleCloudPlatform/kubernetes/pkg/apiserver.func·050()
	/home/travis/gopath/src/github.com/GoogleCloudPlatform/kubernetes/_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/apiserver/handlers_test.go:80 +0xc4
created by github.com/GoogleCloudPlatform/kubernetes/pkg/apiserver.TestMaxInFlight
	/home/travis/gopath/src/github.com/GoogleCloudPlatform/kubernetes/_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/apiserver/handlers_test.go:81 +0x371
goroutine 1 [chan receive]:
testing.RunTests(0x12d66e0, 0x17e5440, 0x43, 0x43, 0xc2080f0800)
	/usr/local/go/src/testing/testing.go:556 +0xdc3
testing.(*M).Run(0xc2080d0be0, 0x2192720)
	/usr/local/go/src/testing/testing.go:485 +0xe1
main.main()
	github.com/GoogleCloudPlatform/kubernetes/pkg/apiserver/_test/_testmain.go:254 +0x382
goroutine 60 [select]:
net/http.(*persistConn).writeLoop(0xc20808a160)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 17 [syscall, locked to thread]:
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2232 +0x1
goroutine 7 [chan receive]:
github.com/golang/glog.(*loggingT).flushDaemon(0x2187b60)
	/home/travis/gopath/src/github.com/GoogleCloudPlatform/kubernetes/Godeps/_workspace/src/github.com/golang/glog/glog.go:879 +0x99
created by github.com/golang/glog.init·1
	/home/travis/gopath/src/github.com/GoogleCloudPlatform/kubernetes/Godeps/_workspace/src/github.com/golang/glog/glog.go:410 +0x2ca
goroutine 59 [select]:
net/http.(*persistConn).readLoop(0xc20808a160)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 495 [select]:
net/http.(*persistConn).readLoop(0xc208770370)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 670 [select]:
net/http.(*persistConn).readLoop(0xc208770e70)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 690 [select]:
net/http.(*persistConn).writeLoop(0xc208770fd0)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 53 [select]:
net/http.(*persistConn).writeLoop(0xc2080e1290)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 50 [select]:
net/http.(*persistConn).writeLoop(0xc2080e11e0)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 31 [select]:
net/http.(*persistConn).writeLoop(0xc2080e1130)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 28 [select]:
net/http.(*persistConn).writeLoop(0xc2080e1080)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 25 [select]:
net/http.(*persistConn).writeLoop(0xc2080e0b00)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 22 [select]:
net/http.(*persistConn).writeLoop(0xc2080e09a0)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 21 [select]:
net/http.(*persistConn).readLoop(0xc2080e09a0)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 24 [select]:
net/http.(*persistConn).readLoop(0xc2080e0b00)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 27 [select]:
net/http.(*persistConn).readLoop(0xc2080e1080)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 30 [select]:
net/http.(*persistConn).readLoop(0xc2080e1130)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 33 [select]:
net/http.(*persistConn).readLoop(0xc2080e11e0)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 52 [select]:
net/http.(*persistConn).readLoop(0xc2080e1290)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 775 [IO wait]:
net.(*pollDesc).Wait(0xc2087678e0, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:84 +0x63
net.(*pollDesc).WaitRead(0xc2087678e0, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:89 +0x51
net.(*netFD).Read(0xc208767880, 0xc20841f000, 0x1000, 0x1000, 0x0, 0x7fa167689008, 0xc208775288)
	/usr/local/go/src/net/fd_unix.go:242 +0x4b3
net.(*conn).Read(0xc2080b65d8, 0xc20841f000, 0x1000, 0x1000, 0xffffffff, 0x0, 0x0)
	/usr/local/go/src/net/net.go:121 +0x125
net.(*TCPConn).Read(0xc2080b65d8, 0xc20841f000, 0x1000, 0x1000, 0xc208774e50, 0x0, 0x0)
	<autogenerated>:36 +0x86
net/http.(*liveSwitchReader).Read(0xc208330908, 0xc20841f000, 0x1000, 0x1000, 0x1040eb0, 0x0, 0x0)
	/usr/local/go/src/net/http/server.go:214 +0xcf
io.(*LimitedReader).Read(0xc208755780, 0xc20841f000, 0x1000, 0x1000, 0x30005, 0x0, 0x0)
	/usr/local/go/src/io/io.go:408 +0x18a
bufio.(*Reader).fill(0xc2086009c0)
	/usr/local/go/src/bufio/bufio.go:97 +0x382
bufio.(*Reader).ReadSlice(0xc2086009c0, 0xa, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/bufio/bufio.go:295 +0x60c
bufio.(*Reader).ReadLine(0xc2086009c0, 0x0, 0x0, 0x0, 0x465800, 0x0, 0x0)
	/usr/local/go/src/bufio/bufio.go:324 +0x7c
net/textproto.(*Reader).readLineSlice(0xc20862a480, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/textproto/reader.go:55 +0xc0
net/textproto.(*Reader).ReadLine(0xc20862a480, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/textproto/reader.go:36 +0x5d
net/http.ReadRequest(0xc2086009c0, 0xc208442ea0, 0x0, 0x0)
	/usr/local/go/src/net/http/request.go:598 +0xdc
net/http.(*conn).readRequest(0xc2083308c0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/server.go:586 +0x3dd
net/http.(*conn).serve(0xc2083308c0)
	/usr/local/go/src/net/http/server.go:1162 +0x976
created by net/http.(*Server).Serve
	/usr/local/go/src/net/http/server.go:1751 +0x3cd
goroutine 68 [select]:
net/http.(*persistConn).readLoop(0xc2080e1340)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 69 [select]:
net/http.(*persistConn).writeLoop(0xc2080e1340)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 72 [select]:
net/http.(*persistConn).readLoop(0xc2080e13f0)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 73 [select]:
net/http.(*persistConn).writeLoop(0xc2080e13f0)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 76 [select]:
net/http.(*persistConn).readLoop(0xc2080e1550)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 77 [select]:
net/http.(*persistConn).writeLoop(0xc2080e1550)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 673 [select]:
net/http.(*persistConn).readLoop(0xc208770fd0)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 80 [select]:
net/http.(*persistConn).readLoop(0xc2080e1600)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 81 [select]:
net/http.(*persistConn).writeLoop(0xc2080e1600)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 84 [select]:
net/http.(*persistConn).readLoop(0xc2080e16b0)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 85 [select]:
net/http.(*persistConn).writeLoop(0xc2080e16b0)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 87 [select]:
net/http.(*persistConn).readLoop(0xc2080e1760)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 88 [select]:
net/http.(*persistConn).writeLoop(0xc2080e1760)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 91 [select]:
net/http.(*persistConn).readLoop(0xc2080e1810)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 92 [select]:
net/http.(*persistConn).writeLoop(0xc2080e1810)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 698 [select]:
net/http.(*persistConn).readLoop(0xc2087714a0)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 95 [select]:
net/http.(*persistConn).readLoop(0xc2080e1970)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 96 [select]:
net/http.(*persistConn).writeLoop(0xc2080e1970)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 104 [select]:
net/http.(*persistConn).readLoop(0xc2080e1c30)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 99 [select]:
net/http.(*persistConn).readLoop(0xc2080e1a20)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 100 [select]:
net/http.(*persistConn).writeLoop(0xc2080e1a20)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 102 [select]:
net/http.(*persistConn).readLoop(0xc2080e1b80)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 103 [select]:
net/http.(*persistConn).writeLoop(0xc2080e1b80)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 105 [select]:
net/http.(*persistConn).writeLoop(0xc2080e1c30)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 183 [select]:
net/http.(*persistConn).writeLoop(0xc208083970)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 182 [select]:
net/http.(*persistConn).readLoop(0xc208083970)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 460 [select]:
net/http.(*persistConn).writeLoop(0xc208460790)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 62 [select]:
net/http.(*persistConn).readLoop(0xc20808a210)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 63 [select]:
net/http.(*persistConn).writeLoop(0xc20808a210)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 671 [select]:
net/http.(*persistConn).writeLoop(0xc208770e70)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 110 [select]:
net/http.(*persistConn).readLoop(0xc2080e1ce0)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 111 [select]:
net/http.(*persistConn).writeLoop(0xc2080e1ce0)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 113 [select]:
net/http.(*persistConn).readLoop(0xc2080e1d90)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 114 [select]:
net/http.(*persistConn).writeLoop(0xc2080e1d90)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 173 [select]:
net/http.(*persistConn).writeLoop(0xc2080e1ad0)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 380 [select]:
net/http.(*persistConn).writeLoop(0xc2087713f0)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 131 [select]:
net/http.(*persistConn).readLoop(0xc20808a370)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 132 [select]:
net/http.(*persistConn).writeLoop(0xc20808a370)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 162 [select]:
net/http.(*persistConn).writeLoop(0xc208082210)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 145 [select]:
net/http.(*persistConn).readLoop(0xc208082210)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 430 [select]:
net/http.(*persistConn).writeLoop(0xc208082580)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 172 [select]:
net/http.(*persistConn).readLoop(0xc2080e1ad0)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 493 [select]:
net/http.(*persistConn).readLoop(0xc2087702c0)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 429 [select]:
net/http.(*persistConn).readLoop(0xc208082580)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 165 [select]:
net/http.(*persistConn).writeLoop(0xc2080e0f20)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 164 [select]:
net/http.(*persistConn).readLoop(0xc2080e0f20)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 459 [select]:
net/http.(*persistConn).readLoop(0xc208460790)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 478 [select]:
net/http.(*persistConn).writeLoop(0xc208770000)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 185 [select]:
net/http.(*persistConn).writeLoop(0xc2080822c0)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 184 [select]:
net/http.(*persistConn).readLoop(0xc2080822c0)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 477 [select]:
net/http.(*persistConn).readLoop(0xc208770000)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 653 [select]:
net/http.(*persistConn).writeLoop(0xc208461ce0)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 652 [select]:
net/http.(*persistConn).readLoop(0xc208461ce0)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 542 [select]:
net/http.(*persistConn).writeLoop(0xc208771e40)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 188 [select]:
net/http.(*persistConn).writeLoop(0xc2080824d0)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 187 [select]:
net/http.(*persistConn).readLoop(0xc2080824d0)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 483 [select]:
net/http.(*persistConn).readLoop(0xc208082790)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 699 [select]:
net/http.(*persistConn).writeLoop(0xc2087714a0)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 425 [select]:
net/http.(*persistConn).writeLoop(0xc208082420)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 484 [select]:
net/http.(*persistConn).writeLoop(0xc208082790)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 191 [select]:
net/http.(*persistConn).writeLoop(0xc208082630)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 190 [select]:
net/http.(*persistConn).readLoop(0xc208082630)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 637 [select]:
net/http.(*persistConn).readLoop(0xc2087704d0)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 424 [select]:
net/http.(*persistConn).readLoop(0xc208082420)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 494 [select]:
net/http.(*persistConn).writeLoop(0xc2087702c0)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 210 [select]:
net/http.(*persistConn).writeLoop(0xc2080e0fd0)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 193 [select]:
net/http.(*persistConn).readLoop(0xc2080e0fd0)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 496 [select]:
net/http.(*persistConn).writeLoop(0xc208770370)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 473 [select]:
net/http.(*persistConn).writeLoop(0xc2087706e0)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 411 [select]:
net/http.(*persistConn).readLoop(0xc208083ce0)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 213 [select]:
net/http.(*persistConn).writeLoop(0xc2080e18c0)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 212 [select]:
net/http.(*persistConn).readLoop(0xc2080e18c0)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 513 [select]:
net/http.(*persistConn).writeLoop(0xc208082bb0)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 355 [select]:
net/http.(*persistConn).writeLoop(0xc2080e14a0)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 379 [select]:
net/http.(*persistConn).readLoop(0xc2087713f0)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 512 [select]:
net/http.(*persistConn).readLoop(0xc208082bb0)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 216 [select]:
net/http.(*persistConn).writeLoop(0xc20808a420)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 215 [select]:
net/http.(*persistConn).readLoop(0xc20808a420)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 529 [select]:
net/http.(*persistConn).writeLoop(0xc20808a630)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 541 [select]:
net/http.(*persistConn).readLoop(0xc208771e40)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 219 [select]:
net/http.(*persistConn).writeLoop(0xc20808a580)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 218 [select]:
net/http.(*persistConn).readLoop(0xc20808a580)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 638 [select]:
net/http.(*persistConn).writeLoop(0xc2087704d0)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 528 [select]:
net/http.(*persistConn).readLoop(0xc20808a630)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 395 [select]:
net/http.(*persistConn).writeLoop(0xc2087718c0)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 251 [select]:
net/http.(*persistConn).readLoop(0xc2087700b0)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 252 [select]:
net/http.(*persistConn).writeLoop(0xc2087700b0)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 443 [select]:
net/http.(*persistConn).readLoop(0xc208771760)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 394 [select]:
net/http.(*persistConn).readLoop(0xc2087718c0)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 401 [select]:
net/http.(*persistConn).writeLoop(0xc208082370)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 400 [select]:
net/http.(*persistConn).readLoop(0xc208082370)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 472 [select]:
net/http.(*persistConn).readLoop(0xc2087706e0)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 412 [select]:
net/http.(*persistConn).writeLoop(0xc208083ce0)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 444 [select]:
net/http.(*persistConn).writeLoop(0xc208771760)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 354 [select]:
net/http.(*persistConn).readLoop(0xc2080e14a0)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 574 [select]:
net/http.(*persistConn).readLoop(0xc2084604d0)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 579 [select]:
net/http.(*persistConn).writeLoop(0xc208460000)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 578 [select]:
net/http.(*persistConn).readLoop(0xc208460000)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 665 [select]:
net/http.(*persistConn).writeLoop(0xc208770a50)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 575 [select]:
net/http.(*persistConn).writeLoop(0xc2084604d0)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 664 [select]:
net/http.(*persistConn).readLoop(0xc208770a50)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 773 [IO wait]:
net.(*pollDesc).Wait(0xc208767720, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:84 +0x63
net.(*pollDesc).WaitRead(0xc208767720, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:89 +0x51
net.(*netFD).accept(0xc2087676c0, 0x0, 0x7fa167689008, 0xc208775238)
	/usr/local/go/src/net/fd_unix.go:419 +0x4bd
net.(*TCPListener).AcceptTCP(0xc2080b6590, 0xc2083308c0, 0x0, 0x0)
	/usr/local/go/src/net/tcpsock_posix.go:234 +0x80
net.(*TCPListener).Accept(0xc2080b6590, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/tcpsock_posix.go:244 +0x5a
net/http/httptest.(*historyListener).Accept(0xc20862a330, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/httptest/server.go:48 +0xa1
net/http.(*Server).Serve(0xc208600660, 0x7fa167692668, 0xc20862a330, 0x0, 0x0)
	/usr/local/go/src/net/http/server.go:1728 +0xa3
created by net/http/httptest.(*Server).Start
	/usr/local/go/src/net/http/httptest/server.go:109 +0x4d7
goroutine 720 [select]:
net/http.(*persistConn).readLoop(0xc208771b80)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 721 [select]:
net/http.(*persistConn).writeLoop(0xc208771b80)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 772 [select]:
net/http.(*persistConn).roundTrip(0xc208460420, 0xc208775190, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:1082 +0xa69
net/http.(*Transport).RoundTrip(0xc208078000, 0xc208442dd0, 0x6, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:235 +0x6b8
net/http.send(0xc208442dd0, 0x7fa1676893b8, 0xc208078000, 0xc2087556c0, 0x0, 0x0)
	/usr/local/go/src/net/http/client.go:219 +0x6f3
net/http.(*Client).send(0x2187600, 0xc208442dd0, 0x16, 0x0, 0x0)
	/usr/local/go/src/net/http/client.go:142 +0x201
net/http.(*Client).doFollowingRedirects(0x2187600, 0xc208442dd0, 0x12d6880, 0x0, 0x0, 0x0)
	/usr/local/go/src/net/http/client.go:367 +0xf18
net/http.(*Client).Do(0x2187600, 0xc208442dd0, 0xc208755660, 0x0, 0x0)
	/usr/local/go/src/net/http/client.go:174 +0xc4
github.com/GoogleCloudPlatform/kubernetes/pkg/apiserver.TestReadOnly(0xc2081be2d0)
	/home/travis/gopath/src/github.com/GoogleCloudPlatform/kubernetes/_output/local/go/src/github.com/GoogleCloudPlatform/kubernetes/pkg/apiserver/handlers_test.go:141 +0x45f
testing.tRunner(0xc2081be2d0, 0x17e5980)
	/usr/local/go/src/testing/testing.go:447 +0x134
created by testing.RunTests
	/usr/local/go/src/testing/testing.go:555 +0xd4e
goroutine 776 [IO wait]:
net.(*pollDesc).Wait(0xc208767870, 0x72, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:84 +0x63
net.(*pollDesc).WaitRead(0xc208767870, 0x0, 0x0)
	/usr/local/go/src/net/fd_poll_runtime.go:89 +0x51
net.(*netFD).Read(0xc208767810, 0xc2085e7000, 0x1000, 0x1000, 0x0, 0x7fa167689008, 0xc2087752b8)
	/usr/local/go/src/net/fd_unix.go:242 +0x4b3
net.(*conn).Read(0xc2080b65e0, 0xc2085e7000, 0x1000, 0x1000, 0x5f48fd, 0x0, 0x0)
	/usr/local/go/src/net/net.go:121 +0x125
net.(*TCPConn).Read(0xc2080b65e0, 0xc2085e7000, 0x1000, 0x1000, 0x101000, 0x0, 0x0)
	<autogenerated>:36 +0x86
net/http.noteEOFReader.Read(0x7fa167692728, 0xc2080b65e0, 0xc208460478, 0xc2085e7000, 0x1000, 0x1000, 0x109a7b0, 0x0, 0x0)
	/usr/local/go/src/net/http/transport.go:1270 +0x7c
net/http.(*noteEOFReader).Read(0xc2087557e0, 0xc2085e7000, 0x1000, 0x1000, 0x5f9a16, 0x0, 0x0)
	<autogenerated>:125 +0xfc
bufio.(*Reader).fill(0xc208600a20)
	/usr/local/go/src/bufio/bufio.go:97 +0x382
bufio.(*Reader).Peek(0xc208600a20, 0x1, 0x0, 0x0, 0x0, 0x0, 0x0)
	/usr/local/go/src/bufio/bufio.go:132 +0x199
net/http.(*persistConn).readLoop(0xc208460420)
	/usr/local/go/src/net/http/transport.go:842 +0xff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 756 [select]:
net/http.(*persistConn).readLoop(0xc208460160)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 757 [select]:
net/http.(*persistConn).writeLoop(0xc208460160)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 760 [select]:
net/http.(*persistConn).readLoop(0xc208460210)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 761 [select]:
net/http.(*persistConn).writeLoop(0xc208460210)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 777 [select]:
net/http.(*persistConn).writeLoop(0xc208460420)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
goroutine 769 [select]:
net/http.(*persistConn).readLoop(0xc208460370)
	/usr/local/go/src/net/http/transport.go:928 +0xec9
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:660 +0x10d4
goroutine 770 [select]:
net/http.(*persistConn).writeLoop(0xc208460370)
	/usr/local/go/src/net/http/transport.go:945 +0x4ff
created by net/http.(*Transport).dialConn
	/usr/local/go/src/net/http/transport.go:661 +0x10f1
FAIL	github.com/GoogleCloudPlatform/kubernetes/pkg/apiserver	2.694s
```

--- a/pkg/apiserver/handlers_test.go
+++ b/pkg/apiserver/handlers_test.go
@@ -43,6 +43,7 @@ func expectHTTP(url string, code int, t *testing.T) {
 	r, err := http.Get(url)
 	if err != nil {
 		t.Errorf("unexpected error: %v", err)
+		return
 	}
 	if r.StatusCode != code {
 		t.Errorf("unexpected response: %v", r.StatusCode)

